File: intellij-plugin/src/main/kotlin/com/ezevillo/gitreview/domain/InstallHint.kt
package com.ezevillo.gitreview.domain

/** Keep in sync with contracts/client-product-surface.yaml. */
const val NPM_INSTALL_CMD: String = "npm install -g git-review-workflow"
const val NPM_UPDATE_CMD: String = "npm install -g git-review-workflow@latest"

enum class CliInstallKind {
    INSTALL, UPDATE,
}

fun npmCommandFor(kind: CliInstallKind): String =
    if (kind == CliInstallKind.UPDATE) NPM_UPDATE_CMD else NPM_INSTALL_CMD

File: visualstudio-extension/src/GitReview.Domain/InstallHint.cs
namespace GitReview.Domain;

/// <summary>Keep in sync with contracts/client-product-surface.yaml.</summary>
public static class InstallHint
{
    public const string NpmInstallCmd = "npm install -g git-review-workflow";
    public const string NpmUpdateCmd = "npm install -g git-review-workflow@latest";

    public enum CliInstallKind
    {
        Install,
        Update,
    }

    public static string NpmCommandFor(CliInstallKind kind) =>
        kind == CliInstallKind.Update ? NpmUpdateCmd : NpmInstallCmd;
}
